### PR TITLE
goomba

### PR DESCRIPTION
**Pull request by nanaian**

### CODE REVIEW

File: src/battle/area_kmr_part_1/battles.c [hunk 1]
```diff
@@ -6,41 +6,41 @@
...
-    { goomba, .position = 2, .priority = 8 },
+    { &goomba, .position = 0, .priority = 10 },
+    { &goomba, .position = 1, .priority = 9 },
+    { &goomba, .position = 2, .priority = 8 },
 };
 
 Formation formation3 = {
-    { goomba, .position = 1, .priority = 10 },
+    { &goomba, .position = 1, .priority = 10 },
     { paragoomba, .position = 6, .priority = 9 },
```
Comment (pixel-stuck) [inline src/battle/area_kmr_part_1/battles.c:25]: this is the first one I noticed this on, but there are others; why do only the goomba's have the & on them?

File: src/battle/actor/goomba.c [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+s32 statusTable_8021961C[] = {
+    0x00000001, 0x00000000, 0x00000002, 0x00000000, 0x00000006, 0x00000064, 0x00000009, 0x00000064,
+    0x00000007, 0x00000064, 0x00000004, 0x00000064, 0x00000003, 0x00000064, 0x0000000B, 0x00000064,
+    0x00000005, 0x00000064, 0x0000000A, 0x00000064, 0x00000008, 0x00000064, 0x0000001F, 0x00000000,
+    0x00000020, 0x00000000, 0x00000025, 0x00000000, 0x00000022, 0x00000000, 0x00000024, 0x00000000,
+    0x00000023, 0x00000000, 0x00000021, 0x00000000, 0x00000026, 0x00000000, 0x00000027, 0x00000000,
+    0x00000029, 0x00000000, 0x00000000,
+};
+
+// 432068-43208C (VRAM: 802196C8)
```
Comment (nanaian) [inline src/battle/actor/goomba.c:33]: @ethteck @pixel-stuck do we want to include these after data is matched or should I delete them

File: src/battle/area_kmr_part_1/battles.c (hunk 1 above)
Comment (pixel-stuck) [reply src/battle/area_kmr_part_1/battles.c:25]: hm. In that case, can you extern them properly, so this is consistent?

File: src/battle/actor/goomba.c (hunk 2 above)
Comment (pixel-stuck) [reply src/battle/actor/goomba.c:54]: They should be deleted imo.

Comment (ethteck) [reply src/battle/actor/goomba.c:54]: yeah I think we're good. We can use the map file / tools if we need to locate this stuff

File: src/battle/actor/goomba.c [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+    Debuff_END,
+};
+
+// 432068-43208C (VRAM: 802196C8)
+ActorPartDesc goomba_parts[] = {
+    {
+        .flags = 0x00800000,
+        .index = 1,
+        .posOffset = { 0, 0, 0 },
+        .targetOffset = { 0, 0x14 },
```
Comment (ethteck) [inline src/battle/actor/goomba.c:60]: decimal for offsets

File: src/battle/actor/goomba.c [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+    SetIdleAnimations(ActorID_SELF, 1, goomba_anims_running);
+    SetIdleGoal(ActorID_SELF, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    IdleRunToGoal(ActorID_SELF, 0);
+    SetIdleAnimations(ActorID_SELF, 1, goomba_anims);
+    loop 80 {
+2:
+        GetStatusFlags(ActorID_SELF, SI_VAR(1));
+        if (SI_VAR(1) & 0x35D000) {
+            sleep 1;
+            goto 2;
```
Comment (ethteck) [inline src/battle/actor/goomba.c:173]: go22 lol

File: src/battle/actor/goomba.c [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+        }
+        sleep 1;
+    }
+    goto 10;
+});
+
+// 432474-432B34 (VRAM: 80219AD4)
```
Comment (ethteck) [inline src/battle/actor/goomba.c:180]: delete comments

File: src/battle/actor/goomba.c [hunk 6]
```diff
@@ -0,0 +1,488 @@
...
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, pain);
+            await DoSpinSmashHit;
+            SI_VAR(0) = c 0x1;
+            SI_VAR(1) = c NPC_ANIM(goomba, normal, dead);
+            await DoDeath;
+            return;
+        }
+        == Event_SHOCK_HIT {
+            SI_VAR(0) = c 0x1;
```
Comment (ethteck) [inline src/battle/actor/goomba.c:223]: probably just 1 instead of 0x1 (also relevant elsewhere)

**Review by ethteck — CHANGES_REQUESTED** [review]
nice! just a couple things and ye

File: src/battle/actor/goomba.c (hunk 6 above)
Comment (nanaian) [reply src/battle/actor/goomba.c:223]: this may actually be a bool, I'm not sure what the DoX scripts expect

**Pull request merged**